9.2/test/run

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+#
+# Test the PostgreSQL image.
+#
+# IMAGE_NAME specifies the name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+
+set -exo nounset
+shopt -s nullglob
+
+IMAGE_NAME=${IMAGE_NAME-openshift/postgresql-92-centos7-candidate}
+
+CIDFILE_DIR=$(mktemp --suffix=postgresql_test_cidfiles -d)
+
+function cleanup() {
+	for cidfile in $CIDFILE_DIR/* ; do
+		CONTAINER=$(cat $cidfile)
+
+		echo "Stopping and removing container $CONTAINER..."
+		docker stop $CONTAINER
+		exit_status=$(docker inspect -f '{{.State.ExitCode}}' $CONTAINER)
+		if [ "$exit_status" != "0" ]; then
+			echo "Dumping logs for $CONTAINER"
+			docker logs $CONTAINER
+		fi
+		docker rm $CONTAINER
+		rm $cidfile
+		echo "Done."
+	done
+	rmdir $CIDFILE_DIR
+}
+trap cleanup EXIT
+
+function get_cid() {
+	local id="$1" ; shift || return 1
+	echo $(cat "$CIDFILE_DIR/$id")
+}
+
+function get_container_ip() {
+	local id="$1" ; shift
+	docker inspect --format='{{.NetworkSettings.IPAddress}}' $(get_cid "$id")
+}
+
+function postgresql_cmd() {
+	docker run --rm -e PGPASSWORD="${PASS}" $IMAGE_NAME psql postgresql://$USER@$CONTAINER_IP:5432/db "$@"
```

Comment:
I was wrong. In fact, what happens is that `assert_login_access` shadows `USER` and `PASS` with local variables of the same name, and because in bash [local variables are visible in the scope of a function *and its children*](http://tldp.org/LDP/abs/html/localvar.html#FTN.AEN18568), `postgresql_cmd` eventually uses those local variables when called from within `assert_login_access`.

Needless to say, that's very very error prone and hard to reason about...